src/rules-factories/dependency-rule.js

```diff
@@ -19,13 +24,30 @@ module.exports = function (ruleMeta, rule, ruleOptions = {}) {
         validateRules(context.settings, options.rules, ruleOptions.validateRules);
       }
 
-      return {
-        ImportDeclaration: (node) => {
-          const dependency = dependencyInfo(node.source.value, node.importKind, context);
+      const additionalDependencyNodesSetting = context.settings[ADDITIONAL_DEPENDENCY_NODES];
+      const additionalDependencyNodes = (
+        isArray(additionalDependencyNodesSetting) ? additionalDependencyNodesSetting : []
+      ).map((dependencyNode) => {
+        if (isString(dependencyNode)) {
+          return PREDEFINED_DEPENDENCY_NODES[dependencyNode];
+        }
+
+        return dependencyNode;
+      });
+
+      const dependencyNodes = [PREDEFINED_DEPENDENCY_NODES.import, ...additionalDependencyNodes]
```

Comment:
__issue__: If an user adds `import` as an additional dependency node, it would be added twice (read my comment above about moving it to a different constant).